### PR TITLE
Average Filter

### PR DESCRIPTION
**Pull request by ricdezen**
## Premise
This pr contains a class to compute the sum and average of a set of keys in a Stream.

## Features
**feat(average_filter.py)**:
**+** Added AverageFilter class: single input and output Filter class that computes the sum and average for each atom that passes through.
**test(average_filter_test.py)** Basic tests for AverageFilter

## Picture of a cute animal
So basically I found out that [Sables](https://it.wikipedia.org/wiki/Martes_zibellina) are the cutest things ever.
zibellino1

### CODE REVIEW

File: otri/filtering/filters/average_filter.py [hunk 1]
```diff
@@ -0,0 +1,69 @@
+from ..filter import Filter, Stream, Collection
+from typing import Sequence, Mapping, Dict
+from numbers import Number
+
+
+class AverageFilter(Filter):
```
Comment (CremaLuca) [inline otri/filtering/filters/average_filter.py:6]: Define inputs and outputs of filter in class specification, so it's easier and faster to link them together.

File: otri/filtering/filters/average_filter.py [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+        Parameters:
+            input_stream : Stream
+                A single Stream of atoms, which we assume to be mappings of some sort.
+            keys : Sequence
+                The keys to compute the average of. Values corresponding to these keys
+                should be numbers.
+        '''
+        super().__init__([input_stream],
+                         input_streams_count=1, output_streams_count=1)
+        self.__cnt = {k: 0 for k in keys}
```
Comment (CremaLuca) [inline otri/filtering/filters/average_filter.py:23]: Woah didn't know you could initialise a dict like that

**Comment by ricdezen** [comment]
Note that we might want to split this filter into two parts. A base one that only sums (which will become the sum filter) ad a subclass that also computes the average.

**Review by CremaLuca — APPROVED** [review]
Clean. 8/8

**Pull request merged**